### PR TITLE
Fix/ecdsa err Fix #344

### LINKED ISSUE #344: The address that recovered by the SignVerifyChip does not match the caller_address
### What command(s) is the bug in?

```
cargo test -p zkevm-circuits --features "onephase, enable-sign-verify, reject-eip2718" --release tx_circuit_tests::tx_circuit_1tx_1max_tx  -- --nocapture
```

### Describe the bug

Running a few unit test inside the tx_circuit in `scroll-stable` branch, we can get `VerifyingError`s like 
```
 constraint: Constraint {
       gate: Gate {
       index: 55,
       name: "caller address == sv_address if it's not zero",
    },
    index: 0,
    name: "caller address == sv_address",
},
```

In the branch [`fix/ecdsa_err`](https://github.com/scroll-tech/zkevm-circuits/commit/58e27c8f839e59777c9ed67a6eaa86438a99c7dd) I added an internal assertion that requires the address that is recovered from `SignData`(which is the only data sent to `SignVerifyChip`) equal to `tx.caller_address`. 

https://github.com/scroll-tech/zkevm-circuits/blob/58e27c8f839e59777c9ed67a6eaa86438a99c7dd/zkevm-circuits/src/tx_circuit.rs#L1676-L1683

Therefore, I think there might be some bug inside the SignVerifyChip.

### Concrete steps to reproduce the bug

_No response_

### PR DESCRIPTION
**Pull request by zhenfeizhang**
### Description

[_PR description_]

### #344 

[_link issue here_]

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- PK endieness was not correct.

### Rationale

[_design decisions and extended information_]

### How Has This Been Tested?
```
cargo test -p zkevm-circuits --features "onephase, enable-sign-verify, reject-eip2718" --release tx_cir  -- --nocapture
```

```
test tx_circuit::tx_circuit_tests::tx_circuit_2tx_2max_tx ... FAILED
test tx_circuit::tx_circuit_tests::tx_circuit_bad_address ... FAILED
test tx_circuit::tx_circuit_tests::tx_circuit_0tx_1max_tx ... ok
test tx_circuit::tx_circuit_tests::tx_circuit_1tx_1max_tx ... ok
test tx_circuit::tx_circuit_tests::tx_circuit_1tx_2max_tx ... ok
test tx_circuit::sign_verify::sign_verify_tests::sign_verify ... ok
```

<hr>

## How to fill a PR description 

Please give a concise description of your PR.

The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.

MUST: Reference the issue to resolve

### Single responsability

Is RECOMMENDED to create single responsibility commits, but not mandatory.

Anyway, you MUST enumerate the changes in a unitary way, e.g.

```
This PR contains:
- Cleanup of xxxx, yyyy
- Changed xxxx to yyyy in order to bla bla
- Added xxxx function to ...
- Refactored ....
```

### Design choices

RECOMMENDED to:
- What types of design choices did you face?
- What decisions you have made?
- Any valuable information that could help reviewers to think critically

### CODE REVIEW

**Comment by lispc** [comment]
tx_circuit_2tx_2max_tx failed?   is it expected?

File: circuit-benchmarks/src/bench_params.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub(crate) const DEGREE: usize = 19;
```
Comment (lispc) [inline circuit-benchmarks/src/bench_params.rs:1]: is this change needed?

File: zkevm-circuits/src/tx_circuit.rs [hunk 2]
```diff
@@ -1675,6 +1675,17 @@ impl<F: Field> SubCircuit<F> for TxCircuit<F> {
...
+
+        // assert tx.caller_address == recovered_pk
+        for (sign_data, tx) in keccak_inputs_sign_verify(&sign_datas)
+            .into_iter()
+            .zip(self.txs.iter())
+        {
+            let pk_hash = keccak(&sign_data);
+            let address = pk_hash.to_address();
+            assert_eq!(address, tx.caller_address);
+        }
```
Comment (lispc) [inline zkevm-circuits/src/tx_circuit.rs:1687]: suggest change to log::error! so the test tx_circuit_bad_address will not panic

**Comment by zhenfeizhang** [comment]
```
test tx_circuit::tx_circuit_tests::tx_circuit_2tx_2max_tx ... FAILED
test tx_circuit::tx_circuit_tests::tx_circuit_bad_address ... FAILED
```

because of the assertion 

```
        // assert tx.caller_address == recovered_pk
        for (sign_data, tx) in keccak_inputs_sign_verify(&sign_datas)
            .into_iter()
            .zip(self.txs.iter())
        {
            let pk_hash = keccak(&sign_data);
            let address = pk_hash.to_address();
            assert_eq!(address, tx.caller_address);
        }
```

```

---- tx_circuit::tx_circuit_tests::tx_circuit_2tx_2max_tx stdout ----
thread 'tx_circuit::tx_circuit_tests::tx_circuit_2tx_2max_tx' panicked at 'assertion failed: `(left == right)`
  left: `0x2e67890492c05eebdb86a6a9d2e0dbabd9629623`,
 right: `0x1c4fab018e8d4d88f0812c40651462c8fe689c7c`', zkevm-circuits/src/tx_circuit.rs:1686:13

---- tx_circuit::tx_circuit_tests::tx_circuit_bad_address stdout ----
thread 'tx_circuit::tx_circuit_tests::tx_circuit_bad_address' panicked at 'assertion failed: `(left == right)`
  left: `0xeefca179f40d3b8b3d941e6a13e48835a3af8241`,
 right: `0x1230000000000000000000000000000000000456`', zkevm-circuits/src/tx_circuit.rs:1686:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

shall we keep this check and return an error instead, or to remove this check? @kunxian-xia

File: circuit-benchmarks/src/bench_params.rs (hunk 1 above)
Comment (zhenfeizhang) [reply circuit-benchmarks/src/bench_params.rs]: fixed with https://github.com/scroll-tech/zkevm-circuits/pull/345/commits/89fafcd8034338adf8a5945433adb36c944748c7

File: zkevm-circuits/src/tx_circuit.rs (hunk 2 above)
Comment (zhenfeizhang) [reply zkevm-circuits/src/tx_circuit.rs:1693]: fixed with https://github.com/scroll-tech/zkevm-circuits/pull/345/commits/89fafcd8034338adf8a5945433adb36c944748c7

**Comment by lispc** [comment]
@kunxian-xia can you have another check of tx_circuit_2tx_2max_tx? strange..

**Review by lispc — APPROVED** [review]

File: zkevm-circuits/src/tx_circuit/sign_verify.rs [hunk 3]
```diff
@@ -386,8 +386,8 @@ impl<F: Field> SignVerifyChip<F> {
         // it is fine to use a phase 1 gate here
         let (_pk, _, address) = ecdsa_chip.range.gate.inner_product(
             ctx,
-            &powers_of_256_cells[0..20].to_vec(),
-            &pk_hash_cells[12..].to_vec(),
```
Comment (kunxian-xia) [inline zkevm-circuits/src/tx_circuit/sign_verify.rs:390]: `pk_hash_cells` is in little endian order?

**Review by kunxian-xia — APPROVED** [review]

File: zkevm-circuits/src/tx_circuit.rs [hunk 4]
```diff
@@ -1675,6 +1675,31 @@ impl<F: Field> SubCircuit<F> for TxCircuit<F> {
             .collect::<Result<Vec<SignData>, Error>>()?;
 
         config.load_aux_tables(layouter)?;
+
+        // check if tx.caller_address == recovered_pk
+        let recovered_pks = keccak_inputs_sign_verify(&sign_datas)
+            .into_iter()
+            .enumerate()
+            .filter(|(idx, _)| {
+                // each sign_data produce two inputs for hashing
```
Comment (zhenfeizhang) [inline zkevm-circuits/src/tx_circuit.rs:1684]: cool! and LGTM!

**Pull request merged**